src/qservkafka/storage/votable.py

```diff
@@ -236,6 +239,184 @@ def _encode_char_column(
+
+        Returns
+        -------
+        str
+            The rewritten URL or the original string.
+
+        """
+        if not (value_str and column.requires_url_rewrite):
+            return value_str
+
+        try:
+            base_url = urlparse(str(config.rewrite_base_url))
+            url = urlparse(value_str)
+            return url._replace(netloc=base_url.netloc).geturl()
+        except Exception as e:
+            self._logger.warning(
+                "Unable to rewrite URL", column=column.name, error=str(e)
+            )
+            return value_str
+
+    def _encode_unicode_null(self, column: JobResultColumnType) -> bytes:
+        """Encode null value for unicodeChar column.
+
+        Parameters
+        ----------
+        column
+            Column type definition.
+
+        Returns
+        -------
+        bytes
+            Serialized representation of a null value for unicodeChar.
+        """
+        if column.arraysize and column.arraysize.variable:
+            return _zero_int
+        elif column.arraysize and column.arraysize.limit:
+            max_bytes = column.arraysize.limit * 2
+            rule = str(max_bytes) + "s"
+            return struct.pack(rule, b"")
+        else:
+            return column.datatype.pack("")
+
+    def _encode_unicode_variable_array(
+        self, value_str: str, value: bytes, column: JobResultColumnType
+    ) -> bytes:
+        """Encode variable-length unicodeChar array.
+
+        Parameters
+        ----------
+        value_str
+            String value to encode.
+        value
+            Encoded byte representation of the string.
+        column
+            Column type definition.
+
+        Returns
+        -------
+        bytes
+            Serialized representation of the variable-length unicodeChar array.
+        """
+        if (
+            column.arraysize
+            and column.arraysize.limit
+            and len(value_str) > column.arraysize.limit
+        ):
+            value_str = value_str[: column.arraysize.limit]
+            value = value_str.encode("utf-16-be")
+
+        char_count = len(value) // 2
+        rule = ">I" + str(len(value)) + "s"
+        return struct.pack(rule, char_count, value)
+
+    def _encode_unicode_fixed_array(
+        self, value_str: str, value: bytes, column: JobResultColumnType
+    ) -> bytes:
+        """Encode fixed-length unicodeChar array.
+
+        Parameters
+        ----------
+        value_str
+            String value to encode.
+        value
+            Encoded byte representation of the string.
+        column
+            Column type definition.
+
+        Returns
+        -------
+        bytes
+            Serialized representation of the fixed unicodeChar array.
+        """
+        if not (column.arraysize and column.arraysize.limit):
+            msg = "Fixed array encoding requires arraysize with limit"
+            raise ValueError(msg)
```

Comment:
You don't otherwise use `column` so if you just pass in `column.arraysize.limit` instead of `column`, you can avoid having to make this check.